chapters/memory.txt

```diff
@@ -3096,18 +3096,175 @@ ifdef::VK_KHR_device_group[]
+ifdef::VK_KHR_device_group[]
+  * pname:memory must: not have been allocated with multiple instances
+endif::VK_KHR_device_group[]
+ifdef::VK_EXT_map_memory_placed[]
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags, the
+    <<features-memoryMapPlaced, pname:memoryMapPlaced>> feature must: be
+    enabled
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags, the
+    pname:pNext chain must: include a slink:VkMemoryMapPlacedInfoEXT
+    structure
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is not enabled, pname:offset must: be zero
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is not enabled, pname:size must: be ename:VK_WHOLE_SIZE
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is enabled, pname:offset must: be aligned to an integer multiple of
+    sname:VkPhysicalDeviceMapMemoryPlacedPropertiesEXT::pname:minPlacedMemoryMapAlignment
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is enabled, pname:size must: be ename:VK_WHOLE_SIZE or be aligned to an
+    integer multiple of
+    sname:VkPhysicalDeviceMapMemoryPlacedPropertiesEXT::pname:minPlacedMemoryMapAlignment
+ifdef::VK_EXT_external_memory_host[]
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags, the memory
+    object must: not have been imported from a handle type of
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_ALLOCATION_BIT_EXT or
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_MAPPED_FOREIGN_MEMORY_BIT_EXT
+endif::VK_EXT_external_memory_host[]
+endif::VK_EXT_map_memory_placed[]
+****
+
+include::{generated}/validity/structs/VkMemoryMapInfoEXT.txt[]
+--
+endif::VK_EXT_map_memory2[]
+
+ifdef::VK_EXT_map_memory_placed[]
+[open,refpage='VkMemoryMapPlacedInfoEXT',desc='Structure containing memory map placement parameters',type='structs']
+--
+If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in
+sname:VkMemoryMapInfoEXT::pname:flags and the pname:pNext chain of
+slink:VkMemoryMapInfoEXT includes a sname:VkMemoryMapPlacedInfoEXT
+structure, then that structure specifies the placement address of the
+memory map.
+The implementation will attempt to place the memory map at
+that address.
+If successful, pname:ppData will be set to the same value as
+sname:VkMemoryMapPlacedInfoEXT::pname:pPlacedAddress and
+fname:vkMapMemory2EXT will return ename:VK_SUCCESS.
+If it cannot place the map at the requested address for any
+reason, the memory object is left unmapped and fname:vkMapMemory2EXT will
+return ename:VK_ERROR_MEMORY_MAP_FAILED.
+
+[NOTE]
+.Note
+====
+In order for a placed map to succeed, the client may: first have to reserve
```

Comment:
```suggestion
In order for a placed map to succeed, the client may first have to reserve
```

We don't use normative tags in notes.

Reply:
Although thinking about it - should this be a note? Also is this something we should advertise via a limit if it's a static property, or is this more "if you haven't reserved it, then it might have already been snatched up by another allocation"?

Reply:
Hrm... Looking at the `mmap()` docs, I suspect we may may actually want two flags:

 - `VK_MEMORY_MAP_TRY_PLACED`: Attempts to place the memory map and fails if another mapping currently exists in that range.
 - `VK_MEMORY_MAP_PLACED_REPLACE`: Replaces whatever is at the current location regardless.  This needs a VU that it's not overwritting another Vulkan mapping because we want the "one map at a time" rule to be validatable.

These map to `MAP_FIXED_NOREPLACE` and `MAP_FIXED`, respectively.  Then it would be explicit what happens and I think we could drop this note entirely.  We could also make them different feature bits.

Reply:
It's simple for applications to emulate `VK_MEMORY_MAP_TRY_PLACED` with a `mmap` call with `MAP_FIXED_NOREPLACE` + `VK_MEMORY_MAP_PLACED_REPLACE`. So I see no reason to have `VK_MEMORY_MAP_TRY_PLACED` at all.

Reply:
That's not going to be atomic.  If something else successfully maps between the two calls, the `PLACED_REPLACE` will stomp it.  I don't know if that's a problem or not.

Reply:
Whether a flag is added to Vulkan or not, a component using MMAP_FIXED/PLACE_REPLACE on an address it hasn't already reserved is a potential race condition.  Adding the MAP_TRY_PLACED would only plug one side of that race condition, while doing nothing against the Vulkan mapping being overwritten.

Reply:
Hrm... Maybe I've not woken up yet.  Yes, I think what @DadSchoorse suggested does work.  If `MAP_FIXED_NOREPLACE` succeeds, you've "reserved" the range and can replace it without worrying about stomping.  Ok, I think I"m convinced that the try isn't needed.

Reply:
I've updated the spec to require replacement behavior.  This should get rid of any of that annoying "may have to reserve" nonsense.